presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -546,6 +556,52 @@ private RelationPlan planLateralJoin(Join join, RelationPlan leftPlan, Lateral l
         return new RelationPlan(planBuilder.getRoot(), analysis.getScope(join), outputSymbols);
     }
 
+    private TranslationMap filterExpressionTranslationMap(RelationPlan leftPlan, RelationPlan rightPlan, PlanBuilder leftPlanBuilder, PlanBuilder rightPlanBuilder, Join join)
+    {
+        List<Symbol> rewriterOutputSymbols = ImmutableList.<Symbol>builder()
+                .addAll(leftPlan.getFieldMappings())
+                .addAll(rightPlan.getFieldMappings())
+                .build();
+
+        PlanNode root = new JoinNode(
+                idAllocator.getNextId(),
+                JoinNode.Type.typeConvert(join.getType()),
+                leftPlanBuilder.getRoot(),
+                rightPlanBuilder.getRoot(),
+                ImmutableList.of(),
+                ImmutableList.<Symbol>builder()
+                        .addAll(leftPlanBuilder.getRoot().getOutputSymbols())
+                        .addAll(rightPlanBuilder.getRoot().getOutputSymbols())
+                        .build(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty());
+
+        RelationPlan intermediateRootRelationPlan = new RelationPlan(root, analysis.getScope(join), rewriterOutputSymbols);
+        TranslationMap translationMap = new TranslationMap(intermediateRootRelationPlan, analysis, lambdaDeclarationToSymbolMap);
+        translationMap.setFieldMappings(rewriterOutputSymbols);
+        translationMap.putExpressionMappingsFrom(leftPlanBuilder.getTranslations());
+        translationMap.putExpressionMappingsFrom(rightPlanBuilder.getTranslations());
+        return translationMap;
+    }
+
+    private Expression extractFilterExpression(Join joinNode)
```

Comment:
Similarly, I'd inline this where it's being used. It's very specific to the planning of that type of join (i.e., join + lateral).